### PR TITLE
Fix support for shading native libraries which was broken in b818852c…

### PR DESCRIPTION
**Pull request by normanmaurer**
…db0ee56ce3cf939e2faa538d519baabd.

Motivation:

b818852cdb0ee56ce3cf939e2faa538d519baabd broke support for shading the native libraries in netty as it missed to respect the package prefix that is used when shading.

Modifications:

Correctly respect package prefix for constructor argument and include the used classname when logging that we could not find the constructor.

Result:

Be able to shade native libraries of netty again.

### CODE REVIEW

**Comment by normanmaurer** [comment]
@cchantep please check as well ... thanks!

**Comment by normanmaurer** [comment]
This fix was verified by @cchantep in https://github.com/netty/netty/issues/8090.

**Comment by normanmaurer** [comment]
I will work on a integration test for this one so we will not break it again

File: transport-native-unix-common/src/main/c/netty_unix_socket.c [hunk 1]
```diff
@@ -1010,21 +1010,32 @@ jint netty_unix_socket_JNI_OnLoad(JNIEnv* env, const char* packagePrefix) {
...
+        free(nettyClassName);
+        nettyClassName = NULL;
         // out-of-memory!
         netty_unix_errors_throwOutOfMemoryError(env);
         return JNI_ERR;
     }
-    datagramSocketAddrMethodId = (*env)->GetMethodID(env, datagramSocketAddressClass, "<init>", "(Ljava/lang/String;IILio/netty/channel/unix/DatagramSocketAddress;)V");
+
+    // Respect shading...
+    char parameters[1024] = {0};
```
Comment (johnou) [inline transport-native-unix-common/src/main/c/netty_unix_socket.c]: Wouldn't it be safer to calculate the length instead of using an arbitrary size?

Comment (normanmaurer) [reply transport-native-unix-common/src/main/c/netty_unix_socket.c]: We could... that said if 1024 is not enough I would be really surprised

**Comment by cchantep** [comment]
@normanmaurer do you know whether it's published on snapshot repo ? To check with CI?

File: transport-native-unix-common/src/main/c/netty_unix_socket.c (hunk 1 above)
Comment (johnou) [reply transport-native-unix-common/src/main/c/netty_unix_socket.c]: I was thinking in terms of buffer overflow but I assume snprintf guards against that? couldn't we do a simple check that nettyClassName does not exceed 1024 then throw illegalargumentexception?

**Pull request merged**